README.md

```diff
@@ -116,6 +128,34 @@ When Git 2.18 or higher is not in your PATH, falls back to the REST API to downl
 - [Checkout pull request on closed event](#Checkout-pull-request-on-closed-event)
 - [Push a commit using the built-in token](#Push-a-commit-using-the-built-in-token)
 
+## Fetch only the root files
+
+```yaml
+- uses: actions/checkout@v3
+  with:
+    sparse-checkout: .
+```
+
+## Fetch only the root files and `.github` and `src` folder
+
+```yaml
+- uses: actions/checkout@v3
+  with:
+    sparse-checkout: |
+      .github
+      src
+```
+
+## Fetch only a single file
```

Comment:
That's sounds great! 🚀 

And maybe it has sense to add an example with `lfs`? In order to let know that the fetch of `lfs` can be reduced with the `sparse-checkout` option to 🤔

Reply:
> do you might provide another example for fetch single file that is not on the root dir? ex: few directory deep, etc. also, maybe some examples of using glob pattern, ex checkout all `*.md` files? 😄

That would require non-cone mode, which is deprecated by the Git project: https://git-scm.com/docs/git-sparse-checkout#_internalscone_pattern_set

Seeing as it is deprecated, I would like to avoid encouraging users to use it. Read: I do not want to add documentation how to do this, giving users the impression that they should use this.

> And maybe it has sense to add an example with `lfs`? In order to let know that the fetch of `lfs` can be reduced with the `sparse-checkout` option to 🤔

The first part would be easy, by adding [another LFS step](https://github.com/actions/checkout/blob/f095bcc56b7c2baf48f3ac70d6d6782f4f553222/.github/workflows/test.yml#L75-L86) that simply uses `sparse-checkout: .`.

However, the second part would be a bit tricky because this test requires a branch with LFS objects, and it uses the `test-data/v2/lfs` branch of `actions/checkout` itself. As you can see [here](https://github.com/actions/checkout/tree/test-data/v2/lfs), there is only one LFS file, in the root directory. Therefore, even in sparse checkout mode "all the LFS files" are fetched, i.e. that single one.

It would require adding another branch to the repository, with an additional LFS file in a subdirectory, and then we would still need to figure out a robust way to verify that it was not fetched, most likely relying on implementation details such as the location of the LFS cache (`.git/lfs/objects`) and the cache files' names, which would be fragile because implementation details are not guaranteed to remain unchanged.

So I don't know whether it is worth the effort.